lib/osf-components/addon/components/activity-log/-components/activity-log-display/component.ts

```diff
@@ -0,0 +1,241 @@
+    tag: string;
+    anonymousLink: string;
+}
+
+export default class ActivityLogDisplayComponent extends Component<ActivityLogDisplayArgs> {
+    @service intl!: Intl;
+    @tracked activityDisplay = '';
+    @tracked isLoading = true;
+    @tracked log!: LogModel;
+    user!: UserModel;
+    linkedNode!: NodeModel;
+    linkedRegistration!: RegistrationModel;
+    node!: NodeModel;
+
+    /**
+     * constructor
+     *
+     * @param owner The owner
+     * @param args The injected args
+     */
+    constructor(owner: unknown, args: ActivityLogDisplayArgs) {
+        super(owner, args);
+        this.log = args.log;
+        taskFor(this.loadModels).perform();
+    }
+
+    /**
+     * buildAHrefElement
+     *
+     * @description Abstracted method to build an "a" element
+     *
+     * @param url The url for the href
+     * @param value The text to appear as clickable
+     * @returns An ahref formatted string
+     */
+    private buildAHrefElement(url: string | undefined, value: string): string {
+        url = url || '';
+        return `<a href="${this.toRelativeUrl(url)}">${value}</a>`;
+    }
+
+    /**
+     * Utility function to convert absolute URLs to relative urls
+     * See:
+     *      http://stackoverflow.com/questions/736513/how-do-i-parse-a-url-into-hostname-and-path-in-javascript
+     * This method have no effect on external urls.
+     * @param url {string} url to be converted
+     * @returns {string} converted relative url
+     */
+    private toRelativeUrl(url: string): string {
+        const parser = document.createElement('a');
+        parser.href = url;
+        let relative_url = url;
+        if (window.location.hostname === parser.hostname){
+            relative_url = parser.pathname + parser.search + parser.hash;
+        }
+        return relative_url;
+    }
+
+    /**
+     * buildParam
+     *
+     * @description Abstracted method to build (assembly) all of the necessary translation variables
+     *
+     * @returns A Param model
+     */
+    private buildParam(): ParamModel {
+        return {
+            anonymousLink: this.log.params.anonymousLink ? 'an anonymous' : 'a',
+            fullName:  this.buildFullNameUrl(),
+            license: this.log.params.license,
+            node:  this.buildNodeUrl(),
+            path: this.log.params.path,
+            pathType: this.log.params.pathType,
+            tag: this.log.params.tag,
+            pointer: this.getPointer(),
+            pointerCategory: this.getPointerCategory(),
+        };
+    }
+
+    /**
+     * loadModels
+     *
+     * @description Hydrates all the models before displaying the component
+     *
+     * @returns a void promise
+     */
+    @task
+    @waitFor
+    private async loadModels(): Promise<void> {
+        this.user = await this.log.user;
+        this.linkedNode = await this.log.linkedNode;
+        this.linkedRegistration = await this.log.linkedRegistration;
+        this.node = await this.log.node;
```

Comment:
Any reason why we shouldn't just embed these into the log request? Doing it this way look like it could be upwards of 40 extra requests to the API per page of logs. Update: It looks like lower down in the diff that we are embedding the data. We shouldn't need to away these, because everything should have been returned with the original request. What kind of problems were you running into with this?

Reply:
I recall having to `await` the relationships even when they are `include`d in the initial log request. I think they are initially `Proxy`s, so you still have to `await` them, but EmberData won't make an additional request if they find the records in the store. Would be worth double-checking though